nixos/doc/manual/release-notes/rl-1903.xml

```diff
@@ -331,17 +331,29 @@
    <para>
      The <literal>pam_unix</literal> account module is now loaded with its
      control field set to <literal>required</literal> instead of
-     <literal>sufficient</literal>, so that later pam account modules that
+     <literal>sufficient</literal>, so that later PAM account modules that
      might do more extensive checks are being executed.
      Previously, the whole account module verification was exited prematurely
      in case a nss module provided the account name to
      <literal>pam_unix</literal>.
      The LDAP and SSSD NixOS modules already add their NSS modules when
-     enabled. In case your setup breaks due to some later pam account module
+     enabled. In case your setup breaks due to some later PAM account module
      previosuly shadowed, or failing NSS lookups, please file a bug. You can
      get back the old behaviour by manually setting
      <literal><![CDATA[security.pam.services.<name?>.text]]></literal>.
    </para>
+  </listitem>
+  <listitem>
+   <para>
+     The <literal>pam_unix</literal> password module is now loaded with its
+     control field set to <literal>sufficient</literal> instead of
+     <literal>required</literal>, so that password managed only
+     by later PAM password modules are being executed.
+     Previously, for example, changing an LDAP account's password through PAM
+     was not possible: the whole password module verification
+     was exited prematurely by <literal>pam_unix</literal>,
+     preventing <literal>pam_ldap</literal> to manage the password as it should.
```

Comment:
We should also mention that (and why) we load `password required pam_deny.so` at the end.

Reply:
I've put that `pam_deny` by precaution, since I've seen it a few line above in `auth required pam_deny.so` and in Debian's `/etc/pam.d/common-password` with the comment: "here's the fallback if no module succeeds". But Debian _seems_ to need that `pam_deny` because it is using some jumping over modules with `[success=N, …]` generated controls, cf. `pam.conf(5)`:
```
N (an unsigned integer)
    equivalent to ok with the side effect of jumping over the next N modules in the stack.
```
As far as I can test, appending `password required pam_deny.so` is not needed in current NixOS PAM config: both with or without it, giving a wrong LDAP password gives an authentication failure as expected; but I am no PAM expert.